src/evaluator/mod.rs

```diff
@@ -321,13 +325,30 @@ impl Evaluator {
         }
     }
 
-    fn eval_call_expr(&mut self, func: Box<Expr>, args: Vec<Expr>) -> Object {
+    fn eval_while_expr(
+        &mut self,
+        cond: &Expr,
+        consequence: &BlockStmt,
+    ) -> Option<Object> {
+        let cond = match self.eval_expr(cond) {
+            Some(cond) => cond,
+            None => return None,
+        };
+
+        while Self::is_truthy(cond.clone()) {
```

Comment:
I am confused about this logic, `cond` is immutable, so we have an infinite loop when it evaluates to `true`, right ?

Reply:
yeah it's a bug, maybe it'd been fixed on my latest commit